kolibri/plugins/learn/assets/src/state/actions.js

```diff
@@ -679,37 +683,40 @@ function setAndSaveCurrentExamAttemptLog(store, contentId, itemId, currentAttemp
     content_id: contentId,
     item: itemId,
   });
-  // If the above findModel returned no matching model, then we can do
-  // getModel to get the new model instead.
-  if (!examAttemptLogModel) {
```

Comment:
It sprang from the fact that I had elsewhere implemented caching for models *without* ids - which previously we did not do - in order to fix an issue where we were saving multiple copies of the same resource, because instead of fetching the same Model, we were just creating new Models over and over again when no id had been set yet.

Here, the issue was that I was getting the model, by the undefined id, and hence still getting a model that was cached - so by directly calling `createModel` I ensure that we get a new model, rather than one that might (later, due to a resolved promise) have an id set later.